bson/datetime_ms.py

```diff
@@ -0,0 +1,134 @@
+import calendar
+import datetime
```

Comment:
Missing a top-level docstring:
```
"""Tools for representing the BSON datetime type."""

import calendar
``` 

We also need to:
- add a doc/api/bson/datetime_ms.rst which you can model after doc/api/bson/ecimal128.rst. 
- add datetime_ms to doc/api/bson/index.rst